src/assets/scripts/aircraft/AircraftInstanceModel.js

```diff
@@ -2021,7 +2021,7 @@ export default class Aircraft {
             }
 
             // lock ILS if at the right angle and altitude
-            if ((abs(this.altitude - glideslope_altitude) < glideslope_window)
+            if (this.altitude < (glideslope_altitude + glideslope_window)
                 && (abs(offset_angle) < degreesToRadians(10))
                 && (offset[1] < ils)
```

Comment:
Something I would like to see the codebase moving towards is abstraction of these large logic blocks.

```javascript
canLockILS() {
    return this.altitude < (glideslope_altitude + glideslope_window)
        && (abs(offset_angle) < degreesToRadians(10))
        && (offset[1] < ils);
}
```

which can then be used as:
```javascript
if (this.canLockILS()) {
    ...
}
```

This is obviously just an example and in reality, the above abstraction should go another step further to be `isWithinAngle()` and `isWithinAltitude()` which would then become
```javascript
canLockILS() {
    return this.isWithinAngle() && this.isWIthinAltitude();
}
```

Even though it seems like more code, and it is, is is much more readable.  It is also more portable.  It is also easier to test.